Source/Project64/N64 System/Mips/Memory Virtual Mem.cpp

```diff
@@ -2376,13 +2376,56 @@ bool CMipsMemoryVM::LB_NonMemory(DWORD PAddr, DWORD* Value, bool /*SignExtend*/)
 		}
 #endif
 	}
-//	switch (PAddr & 0xFFF00000)
-//{
-//	default:
+
+	switch (PAddr & 0xFFF00000)
+	{
+	case 0x08000000:
+		if (g_System->m_SaveUsing == SaveChip_Sram)
+		{
+			//Load byte from Sram
+			BYTE tmp = 0;
+			DWORD Address = PAddr - 0x08000000;
+
+			switch (Address & 0x3)
+			{
+			case 0x0: Address += 3; break;
+			case 0x1: Address ++;   break;
+			case 0x2: Address --;   break;
+			case 0x3: Address -= 3; break;
+			default:
+				return false;
+			}

```

Comment:
Just personal suggestion but maybe instead of doing that switch
```c
switch (Address & 0x3)
{
case 0x0: Address += 3; break;
case 0x1: Address ++;   break;
case 0x2: Address --;   break;
case 0x3: Address -= 3; break;
default:
    return false;
}
```
I was thinking it could be more maintainable code without a switch by using this?
```c
Address ^= 0x000003;
```